jobs/build/jenkins-plugins/Jenkinsfile

```diff
@@ -0,0 +1,123 @@
+                    defaultValue: '3.7'
+                ],
+                [
+                    name: 'PLUGIN_LIST',
+                    description: 'List of plugin:version to include, one per line',
+                    $class: 'hudson.model.TextParameterDefinition'
+                ],
+                [
+                    name: 'MAIL_LIST_SUCCESS',
+                    description: 'Success Mailing List',
+                    $class: 'hudson.model.StringParameterDefinition',
+                    defaultValue: 'jupierce@redhat.com,smunilla@redhat.com,cwong@redhat.com,bparees@redhat.com,pep@redhat.com'
+                ],
+                [
+                    name: 'MAIL_LIST_FAILURE',
+                    description: 'Failure Mailing List',
+                    $class: 'hudson.model.StringParameterDefinition',
+                    defaultValue: 'jupierce@redhat.com,smunilla@redhat.com,cwong@redhat.com,bparees@redhat.com,pep@redhat.com'
+                ],
+                [
+                    name: 'TARGET_NODE',
+                    description: 'Jenkins agent node',
+                    $class: 'hudson.model.StringParameterDefinition',
+                    defaultValue: 'openshift-build-1'
+                ],
+            ]
+        ],
+        disableConcurrentBuilds()
+    ]
+)
+
+def mail_success() {
+
+    jenkins_major = JENKINS_VERSION.tokenize('.')[0].toString()
+
+    distgit_link = "http://pkgs.devel.redhat.com/cgit/rpms/jenkins-${jenkins_major}-plugins/?h=rhaos-${OCP_RELEASE}-rhel-7"
+
+    mail(
+        to: "${MAIL_LIST_SUCCESS}",
+        from: "aos-cd@redhat.com",
+        replyTo: 'jupierce@redhat.com',
+        subject: "jenkins plugins RPM updated in dist-git",
+        body: """The Jenkins plugins RPM for OCP ${OCP_RELEASE} has been updated in dist-git:
+${distgit_link}
+
+Jenkins version: ${JENKINS_VERSION}
+
+Plugin list:
+${PLUGIN_LIST}
+
+Plugin RPM update job: ${env.BUILD_URL}
+""");
+}
+
+def mail_failure(err) {
+
+    mail(
+        to: "${MAIL_LIST_FAILURE}",
+        from: "aos-cd@redhat.com",
+        replyTo: 'jupierce@redhat.com',
+        subject: "Error during jenkins plugin RPM update on dist-git",
+        body: """The job to update the jenkins plugins RPM in dist-git encountered an error:
+${err}
+
+Jenkins job: ${env.BUILD_URL}
+""");
+}
+
+node(TARGET_NODE) {
+
+    checkout scm
+
+    def buildlib = load( "pipeline-scripts/buildlib.groovy")
+    buildlib.kinit()       // Sets up credentials for dist-git access
+    buildlib.path_setup()  // Put scripts in env.PATH
+
+    try {
+        stage ("prepare workspace") {
+
+            // Temporary file to write the plugin list to
+            tmpdir = pwd tmp:true
+            plugin_file = "${tmpdir}/jenkins-plugins.txt"
+            // Note that collect-jenkins-plugins.sh has a hardcoded output dir:
+            plugin_dir = "${env.WORKSPACE}/hacks/update-jenkins-plugins/working/hpis"
```

Comment:
We can change the output directory to CWD if it would simplify things (only CD uses this tool, so we won't break anyone). If we stay with absolute path, I think we might want to rm -rf the expected working directory. If we move to a temporary directory, let's try to clean up the directory if we succeed.

Reply:
I'll go the quick/lazy way and leave the hardcoded path within the scripts if that's ok (adding a cleanup at the end)...